### ISSUE TITLE
Default values on first launch

### ISSUE THREAD

**Issue by ghost** [futuretap/InAppSettingsKit#68]
It seems default values in my plist aren't available on a freshly installed version of my app. For example a toggle switch with default value set to YES results in a NO value until the user actually switches the toggle. I guess that's because default values from the plist aren't written to the settings storage upon first launch. If I'm right, what's the best way to deal with this? Thanks!

P.S. is this the right way to read a value?
    [appSettingsViewController.settingsStore boolForKey:@"myKey"]

**Comment by fabiosoft** [comment]
This works like a charm, even with multiple nested plist files :)
```obj-c
- (void)registerDefaultsFromSettingsBundle{
	// standard stored preference values
	NSUserDefaults* defaults = [NSUserDefaults standardUserDefaults];
	// settings files to process
	NSMutableArray *preferenceFiles = [[NSMutableArray alloc] init];
	// begin with Root file
	[preferenceFiles addObject:@"Root"];
	// as other settings files are discovered will be added to preferencesFiles
	// init IASKSettingsReader for current settings file
	IASKSettingsReader *settingsReader = [[IASKSettingsReader alloc]initWithFile:[preferenceFiles lastObject]];
	[preferenceFiles removeLastObject];
	//filter only plist files from the settings bundle
	NSError *error;
	NSArray *fileinbundle = [[NSFileManager defaultManager]
							 contentsOfDirectoryAtURL:[[settingsReader settingsBundle] resourceURL] includingPropertiesForKeys:@[]
							 options: NSDirectoryEnumerationSkipsHiddenFiles|NSDirectoryEnumerationSkipsSubdirectoryDescendants
							 error:&error];
	NSPredicate *plistFilter = [NSPredicate predicateWithFormat:@"pathExtension='plist'"];
	NSArray *settingsPlistFiles = [fileinbundle filteredArrayUsingPredicate:plistFilter];
	for (NSURL *fileURL in settingsPlistFiles) {
		// extract preference specifiers
		NSArray *preferenceSpecifiers = [[NSDictionary dictionaryWithContentsOfURL:fileURL]objectForKey:kIASKPreferenceSpecifiers];
		// process each specifier in the current settings file
		for (NSDictionary *specifier in preferenceSpecifiers) {
			// get type of current specifier
			NSString *type = [specifier objectForKey:kIASKType];
			// need to check child pane specifier for additional file
			if ([type isEqualToString:kIASKPSChildPaneSpecifier]) {
				[preferenceFiles addObject:[specifier objectForKey:kIASKFile]];
			}else {
				// check if current specifier has a default value
				id defaultValue = [specifier objectForKey:kIASKDefaultValue];
				if (defaultValue) {
					// get key from specifier and current stored preference value
					NSString *key = [specifier objectForKey:kIASKKey];
					id value = [defaults objectForKey:key];
					// update preference value with default value if necessary
					if (key && value == nil) {
						[defaults setObject:defaultValue forKey:key];
					}
				}
			}
		}
	}
	// synchronize stored preference values
	[defaults synchronize];
}
```

**Comment by soenkegissel** [comment]
I want to add a comment here.
Still facing an issue with fresh installed apps. I went trought the Github explainations: https://stackoverflow.com/questions/6667911/having-problems-reading-settings-while-using-inappsettingskit and this thread. My default settings get read on first launch and I can print them. But they are not visible in IASKSettingsKit. I added the func to ...didFinishLaunchingWithOptions...
